src/Sulu/Bundle/ContentBundle/Preview/PreviewMessageHandler.php

```diff
@@ -0,0 +1,307 @@
+
+        // webspace key
+        if (!array_key_exists('webspaceKey', $msg)) {
+            throw new MissingParameterException('webspaceKey');
+        }
+        $webspaceKey = $msg['webspaceKey'];
+        $context->set('webspaceKey', $webspaceKey);
+
+        // user id
+        if (!array_key_exists('user', $msg)) {
+            throw new MissingParameterException('user');
+        }
+        $user = $msg['user'];
+        $context->set('user', $user);
+
+        // content uuid
+        if (!array_key_exists('content', $msg)) {
+            throw new MissingParameterException('content');
+        }
+        $contentUuid = $msg['content'];
+        // filter index page
+        if ($contentUuid === 'index') {
+            $startPage = $this->contentMapper->loadStartPage($webspaceKey, $locale);
+            $contentUuid = $startPage->getUuid();
+        }
+        $context->set('content', $contentUuid);
+
+        // init message vars
+        $template = array_key_exists('template', $msg) ? $msg['template'] : null;
+        $data = array_key_exists('data', $msg) ? $msg['data'] : null;
+
+        // start preview
+        $this->preview->start($user, $contentUuid, $webspaceKey, $locale, $data, $template);
+
+        return array(
+            'command' => 'start',
+            'content' => $contentUuid,
+            'msg' => 'OK'
+        );
+    }
+
+    /**
+     * Stop preview session
+     * @param ConnectionInterface $from
+     * @param MessageHandlerContext $context
+     * @return array
+     * @throws PreviewNotStartedException
+     */
+    private function stop(ConnectionInterface $from, MessageHandlerContext $context)
+    {
+        // check context parameters
+        if (!$context->has('user')) {
+            throw new PreviewNotStartedException();
+        }
+
+        // get user id
+        $user = $context->get('user');
+
+        // get session vars
+        $contentUuid = $context->get('content');
+        $locale = $context->get('locale');
+        $webspaceKey = $context->get('webspaceKey');
+
+        // stop preview
+        $this->preview->stop($user, $contentUuid, $webspaceKey, $locale);
+
+        $context->clear();
+
+        return array(
+            'command' => 'start',
+            'content' => $contentUuid,
+            'msg' => 'OK'
+        );
+    }
+
+    /**
+     * Updates properties of current session content
+     * @param ConnectionInterface $from
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     * @return array
+     * @throws PreviewNotStartedException
+     * @throws MissingParameterException
+     */
+    private function update(ConnectionInterface $from, MessageHandlerContext $context, $msg)
+    {
+        // check context parameters
+        if (
+            !$context->has('content') &&
+            !$context->has('locale') &&
+            !$context->has('webspaceKey') &&
+            !$context->has('user')
+        ) {
```

Comment:
yes something like this but it is a bit more than that it holds the request and session informations ...